frontend/src/libraries/explorer-wamp/nodes.ts

```diff
@@ -7,36 +7,78 @@ export interface NodeInfo {
   nodeId: string;
   lastSeen: number;
   lastHeight: number;
-  lastHash: string | null;
-  signature: string | null;
-  agentName: string | null;
-  agentVersion: string | null;
-  agentBuild: string | null;
-  peerCount: string | null;
-  isValidator: boolean | null;
+  lastHash: string;
+  signature: string;
+  agentName: string;
+  agentVersion: string;
+  agentBuild: string;
+  peerCount: string;
+  isValidator: boolean;
+  status: string;
 }
 
 export default class NodesApi extends ExplorerApi {
-  async getNodes(limit: number = 15, endTimestamp?: number) {
+  async getNodes(
+    limit: number = 15,
+    validatorIndicator: string = "validators",
+    endTimestamp?: number
+  ) {
+    let whereClause = `WHERE last_seen > (strftime('%s','now') - 60) * 1000 `;
+    if (endTimestamp) {
+      whereClause += ` AND last_seen < :endTimestamp`;
+    }
+    if (validatorIndicator === "validators") {
+      whereClause += ` AND is_validator = 1 `;
+    }
+    if (validatorIndicator === "non-validators") {
+      whereClause += ` AND is_validator = 0 `;
+    }
     try {
-      return await this.call<NodeInfo[]>("select", [
+      const nodes = await this.call<NodeInfo[]>("select", [
         `SELECT ip_address as ipAddress, moniker, account_id as accountId, node_id as nodeId, signature, 
-                last_seen as lastSeen, last_height as lastHeight, last_hash as lastHash,
-                agent_name as agentName, agent_version as agentVersion, agent_build as agentBuild,
-                peer_count as peerCount, is_validator as isValidator
-                    FROM nodes
-                    ${endTimestamp ? `WHERE last_seen < :endTimestamp` : ""}
-                    ORDER BY last_seen DESC
-                    Limit :limit`,
+        last_seen as lastSeen, last_height as lastHeight, last_hash as lastHash,
+        agent_name as agentName, agent_version as agentVersion, agent_build as agentBuild,
+        peer_count as peerCount, is_validator as isValidator, status
+            FROM nodes
+            ${whereClause}
+            ORDER BY last_seen DESC
+            LIMIT :limit
+        `,
         {
           limit,
-          endTimestamp
+          endTimestamp,
+          validatorIndicator
         }
       ]);
+      return nodes as NodeInfo[];
     } catch (error) {
       console.error("Nodes.getNodes failed to fetch data due to:");
       console.error(error);
       throw error;
     }
   }
+
+  async getTotalValidatorsAndOther() {
```

Comment:
I would call it `getOnlineNodesStats` and return an object with clearly named elements instead of an array.